File: src/main/java/net/gtaun/shoebill/event/vehicle/TrailerUpdateEvent.java
package net.gtaun.shoebill.event.vehicle;

import net.gtaun.shoebill.object.Player;
import net.gtaun.shoebill.object.Vehicle;
import net.gtaun.util.event.Interruptable;

/**
 * Created by marvin on 18.09.14.
 */
public class TrailerUpdateEvent extends VehicleEvent implements Interruptable {
    private Player player;
    private int ret = 1;

    public TrailerUpdateEvent(Vehicle trailer, Player player) {
        super(trailer);
        this.player = player;
    }

    public Player getPlayer() {
        return player;
    }

    @Override
    public void interrupt() {
        super.interrupt();
    }

    public void disallow() {
        this.ret &= 0;
    }

    public int getResponse() {
        return ret;
    }
}

File: src/main/java/net/gtaun/shoebill/event/vehicle/TrailerUpdateEvent.kt
package net.gtaun.shoebill.event.vehicle

import net.gtaun.shoebill.entities.Player
import net.gtaun.shoebill.entities.Vehicle
import net.gtaun.util.event.Disallowable

/**
 * This event is getting called when the trailer of a [Vehicle] is getting updated.
 *
 * @author Marvin Haschker
 */
class TrailerUpdateEvent(trailer: Vehicle, val player: Player) : VehicleEvent(trailer), Disallowable {

    var response = 1
        private set

    override fun disallow() {
        this.response = this.response and 0
        interrupt()
    }

}
